airflow/providers/amazon/provider.yaml

```diff
@@ -91,6 +91,7 @@ dependencies:
   - redshift_connector>=2.0.888
   - sqlalchemy_redshift>=0.8.6
   - asgiref
+  - python3-saml
```

Comment:
Do we want it as dependency or in additional-extras ?

Reply:
What is the difference?

Reply:
to my perspective dependency is something that is crucial to 85%+ of the providers uses. Without it the provider has no meaning (for example `boto3`)
Extra is something that relatively small subset of users need thus must ask for it explicitly.

The motivation for it is that we need to remember that users do custom stuff on top of Airflow.
For example consider that I have some internal libraries that I install with Airflow (generate my own constraint file) the last thing I want is to handle conflicts with library that I don't even use just because it is installed by default.

Reply:
That's very valid and I definitely think `python3-saml` qualifies as additional-extras then

Reply:
Thanks for the learning

Reply:
Tests are failing now because the module is not installed. How can I specify in tests that I need this extra?

Reply:
I assume testing should be done similar to how we test s3fs extra ?

https://github.com/apache/airflow/blob/69bac3fba897f9e7b0af642c97f9af0987a875de/airflow/providers/amazon/aws/fs/s3.py#L48-L56

https://github.com/apache/airflow/blob/4cc98ba6648b4a350f9093c4e22ad6cff87ac5e4/tests/providers/amazon/aws/fs/test_s3.py#L26

Reply:
Which means skipping these tests :/ I dont think there is another solution but this is not ideal

Reply:
How did AWS added aiobotocore testing?
it's also extra

Reply:
Thanks. Very helpful. But I dont think it is worth creating a new CI workflow for just this test. Tests will be skipped then (it is only one test file, I guess it is fine)